### PR TITLE
Fix processing of compacted single-item JSON-LD collections

### PR DESCRIPTION
**Pull request by ClearlyClaire**
When compacted, single-item JSON-LD arrays are just the value, without array wrapping.

### CODE REVIEW

**Review by renchap — DISMISSED** [review]
Do we need a spec to test / document this behaviour?

**Comment by ShadowJonathan** [comment]
Just curious: is this part of the specification?

**Comment by ClearlyClaire** [comment]
> Just curious: is this part of the specification?

Yes, this is part of the JSON-LD spec and is touched upon in <https://www.w3.org/TR/json-ld11/#representing-singular-values-as-arrays>. ActivityStreams doesn't set `@container` for any of the properties, and allows both ordered lists and single items for `items`: https://www.w3.org/TR/activitystreams-vocabulary/#dfn-items

**Comment by ClearlyClaire** [comment]
> Do we need a spec to test / document this behaviour?

Added some tests

File: spec/services/activitypub/fetch_replies_service_spec.rb [hunk 1]
```diff
@@ -34,6 +34,18 @@
 
   describe '#call' do
     context 'when the payload is a Collection with inlined replies' do
+      context 'when there is a single reply, with the array compacted away' do
+        let(:items) { 'http://example.com/self-reply-1' }
+
+        it 'queues the expeced worker' do
```
Comment (renchap) [inline spec/services/activitypub/fetch_replies_service_spec.rb:40]:
```suggestion
        it 'queues the expected worker' do
```

File: spec/services/activitypub/fetch_featured_collection_service_spec.rb [hunk 2]
```diff
@@ -136,6 +151,21 @@
       end
 
       it_behaves_like 'sets pinned posts'
+
+      context 'when there is a single item, with the array compacted away' do
+        let(:items) { ['https://example.com/account/pinned/unknown-reachable'] }
```
Comment (renchap) [inline spec/services/activitypub/fetch_featured_collection_service_spec.rb:156]: Should probably not be an array here

**Review by renchap — CHANGES_REQUESTED** [review]

**Review by renchap — APPROVED** [review]

**Pull request merged**